src/hydra/dataProvider.ts

```diff
@@ -268,22 +299,33 @@ function dataProvider(
 
   const transformReactAdminDataToRequestBody = (
     resource: string,
-    data: object,
-    extraInformation: { hasFileField?: any },
-  ): Promise<any> => {
-    /** @type {Resource} */
+    data: { [key: string]: unknown } | unknown,
```

Comment:
equivalent to `data: unknown`

(that's why you need to cast type, L314)

Reply:
maybe we need to check the type of data to avoid casting (`data as FormData`, `data as { [key: string]: unknown }`)

Reply:
We cannot really check it, we don't really know the type of `data` when `apiResource` is not found.